sortinghat/core/schema.py

```diff
@@ -1133,27 +1138,23 @@ def resolve_organizations(self, info, filters=None,
     @check_auth
     def resolve_teams(self, info, filters=None, page=1,
                       page_size=settings.DEFAULT_GRAPHQL_PAGE_SIZE, **kwargs):
-        query = Team.objects.order_by('name')
-
         if filters:
+            query = Team.objects.order_by('name')
+            if 'organization' in filters:
+                if not ('name' in filters or 'term' in filters) and 'parent' not in filters:
+                    query = Team.get_root_nodes().order_by('name')
+                query = query.filter(
+                    Q(organization__in=Organization.objects.filter(name=filters['organization'])))
+            if 'parent' in filters:
+                query = query.filter(name=filters['parent'])
+                if query:
+                    query = query.first().get_children()
             if 'name' in filters:
                 query = query.filter(name=filters['name'])
             if 'term' in filters:
-                search_term = filters['term']
-                query = query.filter(Q(name__icontains=search_term))
-            if 'organization' in filters:
-                org_search_term = filters['organization']
-                query = query.filter(Q(organization__in=Organization.objects.filter(name__icontains=org_search_term)))
-                if 'parent' in filters:
-                    query = query.filter(name=filters['parent'])
-                    if query:
-                        query = query.first().get_descendants()
-            else:
-                if 'parent' in filters:
-                    query = query.filter(name=filters['parent'], organization=None)
-                    if query:
-                        query = query.first().get_descendants()
-
+                query = query.filter(name__icontains=filters['term'])
+        else:
+            query = Team.get_root_nodes().order_by('name')
```

Comment:
Although I understand it, I think it will be better to add a comment at the beginning of the if block explaining what it does.

Reply:
@sduenas  Done :+1: